backend/src/test/java/com/engineering/thesis/backend/controllerIntegrationTests/AppointmentEPTests.java

```diff
@@ -0,0 +1,191 @@
+                                            .with(Role)
+                            )
+                            .andExpect(status().isOk());
+                    verify(appointmentService, atLeast(1)).create(any(Appointment.class));
+                }
+                if (name == "Invalid") {
+                    this.mockMvc
+                            .perform(
+                                    post("/api/appointment/create")
+                                            .contentType(MediaType.APPLICATION_JSON)
+                                            .content("{\"id\": 3, \"appointmentDate\":\"2014-04-28T16:00:49.455\", \"cost\":1 , \"patient\": {\"id\": 1}, \"doctor\": {\"id\": 1} }")
+                                            .with(Role)
+                            )
+                            .andExpect(status().isForbidden());
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @Test
+    public void deleteAppointmentShouldBePossibleWhenProperRoleIsSelected() throws Exception {
+        RulesMap().forEach((name, Role) -> {
+            try {
+                System.out.println("Performing delete appointment with role: " + name);
+                if (name == "Doctor" || name == "Patient") {
+                    this.mockMvc
+                            .perform(delete("/api/appointment/delete/1").with(Role))
+                            .andExpect(status().isOk());
+                }
+                if (name == "Invalid") {
+                    this.mockMvc
+                            .perform(delete("/api/appointment/delete/1").with(Role))
+                            .andExpect(status().isForbidden());
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @Test
+    public void updateAppointmentShouldBePossibleWhenProperRoleIsSelected() throws Exception {
+        RulesMap().forEach((name, Role) -> {
+            try {
+                System.out.println("Performing update appointment with role: " + name);
+                if (name == "Doctor" || name == "Patient") {
+                    this.mockMvc
+                            .perform(
+                                    put("/api/appointment/update")
+                                            .contentType(MediaType.APPLICATION_JSON)
+                                            .content("{\"id\": 1, \"appointmentDate\":\"2014-04-28T16:00:49.455\", \"cost\":1 , \"patient\": {\"id\": 1}, \"doctor\": {\"id\": 1} }")
+                                            .with(Role)
+                            )
+                            .andExpect(status().isOk());
+                    verify(appointmentService, atLeast(1)).update(any(Appointment.class));
```

Comment:
We check invocation but not value return due to empty body on most of our methods for C and U (works good as well).